### PR TITLE
genpolicy: Allow default env vars and enable workload identity

### PR DESCRIPTION
**Pull request by sprt**

### CODE REVIEW

File: src/tools/genpolicy/data.json [hunk 1]
```diff
@@ -33,6 +33,14 @@
         "Root": {
             "Path": "/run/kata-containers/shared/containers/$(bundle-id)"
         },
+        "Process": {
+            "Env": [
+                "AZURE_CLIENT_ID=[a-f0-9-]+",
+                "AZURE_TENANT_ID=[a-f0-9-]+",
```
Comment (sprt) [inline src/tools/genpolicy/data.json:39]: TODO: Also allow capitals even if not seen in my testing

File: src/tools/genpolicy/src/policy.rs [hunk 2]
```diff
@@ -78,21 +78,24 @@ pub struct KataSpec {
 
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
 pub struct KataProcess {
+    #[serde(default)]
     pub Terminal: bool,
+    #[serde(default)]
     pub User: KataUser,
-
+
```
Comment (ms-mahuber) [inline src/tools/genpolicy/src/policy.rs:85]: let's not introduce whitespaces, tabs or any unnecessary stuff. best to configure VS code to truncate such things automatically.

**Review by BryceDFisher — APPROVED** [review]
My only concern is that as we continue to introduce additional AKS and Kubernetes features, the data.json is going to grow to hold more and more ENV and mount points (and maybe even whole containers in the case of the monitoring sidecars that can be introduced).  I don't think it will be the customer's intent to potentially allow all of these things if the feature isn't requested in their pod yaml.

**Review by danmihai1 — CHANGES_REQUESTED** [review]

File: src/tools/genpolicy/rules.rego [hunk 3]
```diff
@@ -545,6 +545,29 @@ allow_env(policy_process, input_process, sandbox_name) {
...
+
+    print("allow_env_var regex 1: success")
+}
+
+# Allow input env variables that match with request_defaults, after substituting $(sandbox-name).
+allow_env_var(policy_process, input_process, env_var, sandbox_name) {
+    print("allow_env_var regex 2: some allow_env_regex match env_var")
+
+    some policy_var_regex in policy_data.request_defaults.CreateContainerRequest.allow_env_regex
+    policy_regex = replace(policy_var_regex, "$(sandbox-name)", sandbox_name)
```
Comment (danmihai1) [inline src/tools/genpolicy/rules.rego:563]: Do we need this second set of allow rules? 

I don't see any $(sandbox-name) in the current allow_env_regex - do you expect some of them in the future?

**Review by danmihai1 — CHANGES_REQUESTED** [review]

**Review by danmihai1 — APPROVED** [review]

**Pull request merged**